util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -111,9 +111,10 @@ protected Map<Object, ChildLbState> createChildLbMap(ResolvedAddresses resolvedA
     Object policyConfig = resolvedAddresses.getLoadBalancingPolicyConfig();
     for (EquivalentAddressGroup eag : addresses) {
       EquivalentAddressGroup strippedEag = stripAttrs(eag); // keys need to be just addresses
-      ChildLbState childLbState = childLbMap.getOrDefault(strippedEag,
-          createChildLbState(strippedEag, policyConfig, getInitialPicker()));
-      childLbMap.put(strippedEag, childLbState);
+      if (!childLbMap.containsKey(strippedEag)) {
+        childLbMap.put(strippedEag,
+            createChildLbState(strippedEag, policyConfig, getInitialPicker()));
```

Comment:
For a follow-up PR:

`policyConfig` handling here looks broken.
1. This config looks to be the config of the parent, not the child
2. The config is never updated

I think this is probably working because pick first is the only child and PickFirstLoadBalancer is too generous, and doesn't blindly cast. https://github.com/grpc/grpc-java/blob/7d9b76e3304bac28ed3d038f6a5f8d5d8b66103c/core/src/main/java/io/grpc/internal/PickFirstLoadBalancer.java#L64

It'd could be considered okay to use a non-changing config for the moment (although I'd probably discourage it, as adding in those flows later is probably harder, especially if this is supposed to be a general-purpose utility), but passing the parent's config to the child is just asking for confusion.